src/transformers/prompt_utils.py

```diff
@@ -0,0 +1,478 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Prompt configuration class and utilities."""
+
+import copy
+import json
+import os
+import warnings
+from typing import Any, Dict, Optional, Union
+
+from . import __version__
+from .configuration_utils import PretrainedConfig
+from .utils import (
+    PROMPT_CONFIG_NAME,
+    PushToHubMixin,
+    cached_file,
+    download_url,
+    extract_commit_hash,
+    is_remote_url,
+    logging,
+)
+
+
+logger = logging.get_logger(__name__)
+
+
+class PromptConfig(PushToHubMixin):
+    r"""
+    This class holds the prompt configuration used when training a model. This is used to ensure that downstream
+    fine-tuning and generation runs can correctly match the prompt configuration used in training, avoiding silent
+    quality loss due to distribution drift.
+
+    This class contains several 'standard' parameters which are common to most chat-based models.
+    """
+
+    # TODO Fill in docstring
+    def __init__(self, **kwargs):
+        self.role_prefixes = kwargs.pop("chat_role_prefixes", {})
```

Comment:
Any particular reason why the variables are called X, but the corresponding kwargs `chat_X`?